### DISCUSSION TITLE
Why doesn't it report an error?

### DISCUSSION THREAD

**Discussion by w1475963** [python-lsp/python-lsp-server#551] [Q&A]
When I use some undefined variables, it doesn't give me an error.

![IMG_20240415_162116](https://github.com/python-lsp/python-lsp-server/assets/46069701/ef898a9a-7dd4-4e07-b820-1f283a98dff7)

line 15, line 16, line 18, and line 19.

**Comment by gmoutso** [comment]
I am trying to understand this! I am trying to move away from pyright where I have everything configured, anticipating a feature-full pylsp-rope. But I can't even get some of the basics. First, why do I not get unknown module or unknown name in module? None of the following give me any errors
```
import nonsense_module
print(nonsense_module)
from numpy import nonsense_name
print(nonsense_name)
import pandas
pandas.DataFrame().do_nonsense()
```
I have set pylsp.plugins.jedi.environment to a conda environment path and work in emacs eglot. I can autocomplete from my conda environment and I can also complete from my source code but rather bizarely (not having specified my top module starts in a folder called python). That is, I can autocomplete all of the below
```
import python.top_mod.sub_mod
import top_mod.sub_mod
import sub_mod
```
Only the second makes sense.
